src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs

```diff
@@ -3036,117 +3028,152 @@ public static void Main()
 
             verify.VerifyIL("C<T>.Print()", @"
 {
-  // Code size       62 (0x3e)
+  // Code size       43 (0x2b)
   .maxstack  2
-  .locals init (T V_0,
-                T V_1)
+  .locals init (T V_0)
   IL_0000:  ldarg.0
   IL_0001:  call       ""T C<T>.M()""
   IL_0006:  stloc.0
   IL_0007:  ldloca.s   V_0
-  IL_0009:  ldloca.s   V_1
-  IL_000b:  initobj    ""T""
-  IL_0011:  ldloc.1
-  IL_0012:  box        ""T""
-  IL_0017:  brtrue.s   IL_002d
-  IL_0019:  ldobj      ""T""
-  IL_001e:  stloc.1
-  IL_001f:  ldloca.s   V_1
-  IL_0021:  ldloc.1
-  IL_0022:  box        ""T""
-  IL_0027:  brtrue.s   IL_002d
-  IL_0029:  pop
-  IL_002a:  ldnull
-  IL_002b:  br.s       IL_0038
-  IL_002d:  constrained. ""T""
-  IL_0033:  callvirt   ""string object.ToString()""
-  IL_0038:  call       ""void System.Console.WriteLine(string)""
-  IL_003d:  ret
+  IL_0009:  dup
+  IL_000a:  ldobj      ""T""
+  IL_000f:  box        ""T""
+  IL_0014:  brtrue.s   IL_001a
+  IL_0016:  pop
+  IL_0017:  ldnull
+  IL_0018:  br.s       IL_0025
+  IL_001a:  constrained. ""T""
+  IL_0020:  callvirt   ""string object.ToString()""
+  IL_0025:  call       ""void System.Console.WriteLine(string)""
+  IL_002a:  ret
 }");
+
         }
 
         [Fact, WorkItem(40690, "https://github.com/dotnet/roslyn/issues/40690")]
         public void ConditionalAccess_GenericExtension_ValueTuple()
         {
             var source = @"
 using System;
-using System.Collections;
-using System.Collections.Generic;
-using System.Linq;
 
-namespace bug
+public static class Extension
 {
-    public static class Extension
+    public static string GetValue(this object value) => value?.ToString();
+}
+public class Class<T>
+{
+    public (long, T) Data { get; }
+    public Class((long, T) data)
     {
-        public static String GetValue(this object value) => value?.ToString();
+        Data = data;
     }
-
-    public partial class BGen<__T__>
+    internal string Value
     {
-        public (Int64, __T__) Data { get; }
-
-        public BGen((Int64, __T__) data)
+        get
         {
-            this.Data = data;
+            var q2 = Data.Item2?.GetValue();
+            return q2;
         }
+    }
+}
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine(new Class<string>((0, ""abc"")).Value);
```

Comment:
Are there any other scenarios we need to make sure we have tests for? (not a ton are coming to my mind.)

Reply:
I can't think of any :-)